src/asmcnc/skavaUI/screen_job_recovery.py

```diff
@@ -0,0 +1,419 @@
+        feedrate_line = next((s for s in reversed(self.jd.job_gcode[:self.selected_line_index + 1]) if 'F' in s), None)
+        if feedrate_line:
+            self.feed = re.match('\d+',feedrate_line[feedrate_line.find("F")+1:]).group()
+        else:
+            self.feed = "Undefined"
+
+        # Recover most recent position
+        x_line = next((s for s in reversed(self.jd.job_gcode[:self.selected_line_index + 1]) if 'X' in s), None)
+        if x_line:
+            self.pos_x = float(re.split('(X|Y|Z|F|S|I|J|K|G)', x_line)[re.split('(X|Y|Z|F|S|I|J|K|G)', x_line).index('X') + 1])
+        else:
+            self.pos_x = 0.0
+        y_line = next((s for s in reversed(self.jd.job_gcode[:self.selected_line_index + 1]) if 'Y' in s), None)
+        if y_line:
+            self.pos_y = float(re.split('(X|Y|Z|F|S|I|J|K|G)', y_line)[re.split('(X|Y|Z|F|S|I|J|K|G)', y_line).index('Y') + 1])
+        else:
+            self.pos_y = 0.0
+        z_line = next((s for s in reversed(self.jd.job_gcode[:self.selected_line_index + 1]) if 'Z' in s), None)
+        if z_line:
+            self.pos_z = float(re.split('(X|Y|Z|F|S|I|J|K|G)', z_line)[re.split('(X|Y|Z|F|S|I|J|K|G)', z_line).index('Z') + 1])
+        else:
+            self.pos_z = 0.0
+
+        self.pos_label.text = "wX: %s | wY: %s | wZ: %s" % (str(self.pos_x), str(self.pos_y), str(self.pos_z))
+        self.speed_label.text = "F: %s | S: %s" % (str(self.feed), str(self.speed))
+
+    def get_info(self):
+
+        info = ('This screen allows you to choose where to recover your job from.\n\n'
+                'Use the arrows to navigate through the lines of the job file, and select a point to recover the job from. '
+                'The red marker indicates where the stall event happened. We suggest starting a few lines before here.\n\n'
+                'To confirm this start point, use the "GO XY" button. This will move the Z Head over the selected start point. '
+                'You can lower the spindle to check the tool is approximately where you expect it to be in the XY plane. '
+                'This XY position can be fine adjusted in the next screen. \n\n'
+                'Note: the Z position will be automatically calculated when the job recovery starts.\n\n'
+                'Note: when using job recovery for GCode containing G2 & G3 commands, '
+                'the recovery point may be highlighted earlier in the file than actual, due to file streaming protocols.'
+        )
+
+        popup_info.PopupBigInfo(self.sm, self.l, 760, info)
+
+    def go_xy(self):
+        self.m.s.write_command('G90 G0 X%s Y%s' % (self.pos_x, self.pos_y))
+
+    def back_to_home(self):
+        self.jd.reset_recovery()
+        self.sm.current = 'home'
+
+    def next_screen(self):
+        if self.m.state().startswith("Idle"):
+            self.wait_popup = popup_info.PopupWait(self.sm, self.l)
+            self.jd.job_recovery_selected_line = self.selected_line_index + 1
```

Comment:
Was asked to make it so that the selected line is included in the scrape and then have it continue on from the next line. Bit more clear when the position and speed info at the bottom reflects the selected line rather than the one before it I guess

Reply:
does that mean that the line you select to go from isn't actually the line that's executed? It's the next line that's executed?

Reply:
Yep exactly